pkg/edgedictionary/edgedictionary_test.go

```diff
@@ -0,0 +1,444 @@
+			api:        mock.API{UpdateDictionaryFn: updateDictionaryOK},
+			wantOutput: updateDictionaryOutput,
+		},
+		{
+			args:       []string{"dictionary", "update", "-v", "--service-id", "123", "--version", "1", "--name", "oldname", "--new-name", "dict-1"},
+			api:        mock.API{UpdateDictionaryFn: updateDictionaryOK},
+			wantOutput: updateDictionaryOutputVerbose,
+		},
+		{
+			args:      []string{"dictionary", "update", "--service-id", "123", "--version", "1", "--name", "oldname", "--new-name", "dict-1"},
+			api:       mock.API{UpdateDictionaryFn: updateDictionaryError},
+			wantError: errTest.Error(),
+		},
+	} {
+		t.Run(strings.Join(testcase.args, " "), func(t *testing.T) {
+			var (
+				args                           = testcase.args
+				env                            = config.Environment{}
+				file                           = config.File{}
+				appConfigFile                  = "/dev/null"
+				clientFactory                  = mock.APIClient(testcase.api)
+				httpClient                     = http.DefaultClient
+				versioner     update.Versioner = nil
+				in            io.Reader        = nil
+				out           bytes.Buffer
+			)
+			err := app.Run(args, env, file, appConfigFile, clientFactory, httpClient, versioner, in, &out)
+			testutil.AssertErrorContains(t, err, testcase.wantError)
+			testutil.AssertString(t, testcase.wantOutput, out.String())
+		})
+	}
+}
+
+func describeDictionaryOK(i *fastly.GetDictionaryInput) (*fastly.Dictionary, error) {
+	return &fastly.Dictionary{
+		ServiceID:      i.ServiceID,
+		ServiceVersion: i.ServiceVersion,
+		Name:           i.Name,
+		CreatedAt:      testutil.MustParseTimeRFC3339("2001-02-03T04:05:06Z"),
+		WriteOnly:      false,
+		ID:             "456",
+		UpdatedAt:      testutil.MustParseTimeRFC3339("2001-02-03T04:05:07Z"),
+	}, nil
+}
+
+func describeDictionaryOKDeleted(i *fastly.GetDictionaryInput) (*fastly.Dictionary, error) {
+	return &fastly.Dictionary{
+		ServiceID:      i.ServiceID,
+		ServiceVersion: i.ServiceVersion,
+		Name:           i.Name,
+		CreatedAt:      testutil.MustParseTimeRFC3339("2001-02-03T04:05:06Z"),
+		WriteOnly:      false,
+		ID:             "456",
+		UpdatedAt:      testutil.MustParseTimeRFC3339("2001-02-03T04:05:07Z"),
+		DeletedAt:      testutil.MustParseTimeRFC3339("2001-02-03T04:05:08Z"),
+	}, nil
+}
+
+func createDictionaryOK(i *fastly.CreateDictionaryInput) (*fastly.Dictionary, error) {
+	return &fastly.Dictionary{
+		ServiceID:      i.ServiceID,
+		ServiceVersion: i.ServiceVersion,
+		Name:           i.Name,
+		CreatedAt:      testutil.MustParseTimeRFC3339("2001-02-03T04:05:06Z"),
+		WriteOnly:      false,
+		ID:             "456",
+		UpdatedAt:      testutil.MustParseTimeRFC3339("2001-02-03T04:05:07Z"),
+	}, nil
+}
+
+// These two are responses from other packages that will need to update if
+// the responses ever change.
```

Comment:
I'm not sure I quite understood this comment, could you possible rephrase it or clarify for me please.

Reply:
I had to re-read it myself, so I'll clarify the comment and split it so it covers each function. I was attempting to point out that these methods mock fastly api calls that are not part of the local edgedictionary package.

Reply:
Updated. Please let me know if the comments are unclear or unnecessary.